### PR TITLE
freeimage: new package

### PR DESCRIPTION
**Pull request by matz-e**

### CODE REVIEW

File: var/spack/repos/builtin/packages/freeimage/package.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+class Freeimage(MakefilePackage):
+    """FreeImage is an Open Source library project for developers who would like
+       to support popular graphics image formats like PNG, BMP, JPEG, TIFF and
+       others as needed by today's multimedia applications"""
+
+    homepage = "http://freeimage.sourceforge.net/"
+
+    version('3.18.0', 'f8ba138a3be233a3eed9c456e42e2578')
+
+    patch('install_fixes.patch', when='@3.18.0')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/freeimage/package.py]: It looks like you forgot to check in this patch.

File: var/spack/repos/builtin/packages/freeimage/package.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+    def edit(self, spec, prefix):
+        env["DESTDIR"] = prefix
+
+    def url_for_version(self, version):
+        url = "https://downloads.sourceforge.net/project/freeimage/Source%20Distribution/{0}/FreeImage{1}.zip"
+        return url.format(version, str(version).replace('.', ''))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/freeimage/package.py]:
```suggestion
        return url.format(version, version.joined)
```

File: var/spack/repos/builtin/packages/freeimage/package.py (hunk 1 above)
Comment (matz-e) [reply var/spack/repos/builtin/packages/freeimage/package.py]: Oops, sorry. Fixed now.

File: var/spack/repos/builtin/packages/freeimage/package.py (hunk 2 above)
Comment (matz-e) [reply var/spack/repos/builtin/packages/freeimage/package.py]: Handy, did not know about that. Thank you!

**Comment by adamjstewart** [comment]
```
var/spack/repos/builtin/packages/freeimage/package.py:8: [E302] expected 2 blank lines, found 1
```

**Comment by matz-e** [comment]
> ```
> var/spack/repos/builtin/packages/freeimage/package.py:8: [E302] expected 2 blank lines, found 1
> ```

Sorry again. Got some more updates coming, I'll be more diligent.

**Comment by adamjstewart** [comment]
@matz-e more updates after the flake8 fix, or is this ready to merge?

**Comment by matz-e** [comment]
> @matz-e more updates after the flake8 fix, or is this ready to merge?

This is ready to merge. More updates as in other packages to update.

**Pull request merged**